org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -86,6 +93,47 @@ public IResourceSetProvider getResourceSetProvider() {
 		return resourceSetProvider;
 	}
 	
+	/**
+	 * Name of a preference defining the scheduling rule strategy for the builder.
+	 * 
+	 * @since 2.16
+	 */
+	@Beta
+	public static final String PREF_SCHEDULING_RULE = "schedulingrule"; //$NON-NLS-1$
+	/**
+	 * Options for scheduling rules.
+	 * 
+	 * @since 2.16
+	 */
+	public static enum SchedulingOption {
+		WORKSPACE,
+		// all Xtext projects in the workspace
+		ALL_XTEXT_PROJECTS,
+		// the currently building project
+		PROJECT,
+		// null scheduling rule
+		NULL;
+	}
+	
+	private SchedulingOption schedulingOption;
+	
+	public XtextBuilder () {
+		IPreferenceStore preferenceStore = new ScopedPreferenceStore(InstanceScope.INSTANCE,
```

Comment:
am not sure of we should move this to a post constuct `@Inject` annotated method. @szarnekow what do you think?

Reply:
My thoughts on this:
Right now there will be one XtextBuilder per project thus one listener per project. Maybe we should move this preference handling to another class that is annotated with `@Singleton`. This would be injected anyway. Also it would be easier to test if the cached preference is updated after setting a value on the store.

Reply:
:+1:

Reply:
Good idea. I will implement this.

Reply:
Updated